### DISCUSSION TITLE
Widget ressource edit - access data

### DISCUSSION THREAD

**Discussion by Grazulex** [filamentphp/filament#3758] [Help] [answered]
Hello,

I have create a widget to put this in my Edit page.
I try to have access from this to the recaord with this :

```
class CashOverview extends BaseWidget
{
    protected int | string | array $columnSpan = 'full';

    public ?Model $record = null;

    public static function canView(): bool
    {}

}
```

How can I have access to this "$record" from the "canView" function ?

Thanks you

JMS

**Comment by danharrin** [comment, marked as answer]
That's not possible right now as it's a static method

You should use `$this->record` in the `getHeaderWidgets()` and only return that item in the array if you want the widget to display.

**Reply by antoinemy** [reply]
Hello @danharrin,

I do not understand the documentation well and I would like you to explain to me how to use this famous `$this->record` correctly.

Here I have a custom page where I list different data as Line Chart widgets. So far so good, I created my custom widget that I duplicate in a `foreach($records as $record)` but I can't get the data from each `$record` to use them in my methods `getData()` or `getOptions()`

Reading the documentation, I know it's this famous:

```
use Illuminate\Database\Eloquent\Model;
 
public ?Model $record = null;
```

In my custom page :
```
protected function getHeaderWidgets(): array
{
    return [
        TestChart::class,
    ];
}
```
In my TestChart : 
```
public ?Model $record = null;

protected function getData(): array
{
    // Need to use $this->record but it's always null
    // I can't pass the data
}

protected function getOptions(): array
{
    // Need to use $this->record but it's always null
    // I can't pass the data
}
```
But impossible to use it properly, can you enlighten me about this with an example?

Thank's

**Reply by danharrin** [reply]
`$this->record` should work out of the box for widgets that are placed on Edit or View pages, but you have a custom page...

So you need to tell Filament where it should find the record to pass to the widget.

If your custom page has its own `public $record`, you can use this in the custom page view:

```blade
<x-filament::page :widget-data="['record' => $record]">
```

Any data in that `widget-data` array is now available in the widget

**Reply by antoinemy** [reply]
It works perfectly, thank you! I had done something similar but it was DIY ^^'

**Comment by Grazulex** [comment]
Oh...Thanks
In this widget (table one) I list records and I used the same widget in the create and the edit action but I need to filter this list in the edit action with the current record id (parent one). What will be the best solution ?
Thanks

**Reply by danharrin** [reply]
You can access `$this->record` within `getTableQuery()`

**Reply by Grazulex** [reply]
Many thanks..in the best word I need this access to in the CanView fonction (I check if I have records to show or not the widget) but I'm happy so now. thanks you
